In [2]:
import pandas as pd

import matplotlib.pyplot as plt
import seaborn as sns

Here we are importing the raw data from repository. Next step is to see the first head data

In [3]:
telco_customers = pd.read_csv('data/Dataset_ATS_v2.csv')

telco_customers.head()

,gender,SeniorCitizen,Dependents,tenure,PhoneService,MultipleLines,InternetService,Contract,MonthlyCharges,Churn
0,Female,0,No,1,No,No,DSL,Month-to-month,25,Yes
1,Male,0,No,41,Yes,No,DSL,One year,25,No
2,Female,0,Yes,52,Yes,No,DSL,Month-to-month,19,No
3,Female,0,No,1,Yes,No,DSL,One year,76,Yes
4,Male,0,No,67,Yes,No,Fiber optic,Month-to-month,51,No


Based on the table

In [6]:
telco_customers.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7043 entries, 0 to 7042
Data columns (total 10 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   gender           7043 non-null   object
 1   SeniorCitizen    7043 non-null   int64 
 2   Dependents       7043 non-null   object
 3   tenure           7043 non-null   int64 
 4   PhoneService     7043 non-null   object
 5   MultipleLines    7043 non-null   object
 6   InternetService  7043 non-null   object
 7   Contract         7043 non-null   object
 8   MonthlyCharges   7043 non-null   int64 
 9   Churn            7043 non-null   object
dtypes: int64(3), object(7)
memory usage: 550.4+ KB


In [7]:
telco_customers.shape

(7043, 10)

The raw dataset contains: 7043 Columns and 10 Rows.
Next steps we'll check whether there are missing values

In [8]:
telco_customers.isnull().any()

gender             False
SeniorCitizen      False
Dependents         False
tenure             False
PhoneService       False
MultipleLines      False
InternetService    False
Contract           False
MonthlyCharges     False
Churn              False
dtype: bool

Seems like there are no missing value so this is a perfectly clean data. Let's take a look at brief overview of the statistical data

In [10]:
telco_customers.describe()

,SeniorCitizen,tenure,MonthlyCharges
count,7043.000000,7043.000000,7043.000000
mean,0.162147,32.371149,64.758768
std,0.368612,24.559481,30.091650
min,0.000000,0.000000,18.000000
25%,0.000000,9.000000,36.000000
50%,0.000000,29.000000,70.000000
75%,0.000000,55.000000,90.000000
max,1.000000,72.000000,119.000000


Here we only see ``SeniorCitizen``, ``tenure``, and ``MonthlyCharges`` out of 10 variable appears on the table because out of all column only 3 contains numerical values